### ISSUE TITLE
Missing 201-keys in Multi-Status responses when 409 or 500 keys are present

### ISSUE THREAD

**Issue by fredrb** [eu-federation-gateway-service/efgs-federation-gateway#135]
## Describe the bug

When processing a 207 response from EFGS, we can get three status per key: `201`, `409` or `500`. Based on the example from upload controller, I would expect all values to possibly be filled on the response.
```java
// ...
examples = @ExampleObject("{\n  '201': [1,2,5,8,9],\n  '409': [3,4,6,7],\n  '500': [10]\n}"))
```

The result on my side for a 4k keys upload looks something like:
```
{
  '500': [234, 567],
  '201': [],
  '409': [1, 2, 3, 4, 5, 6, ... ] // 1k length array
}
```

## Expected behaviour

I expect that on a 4k keys batch all key indexes can be found on the response body. Meaning that if 1k keys are in `409` status, no keys in `500`, we should have 3k keys on `201` status. 

## Steps to reproduce the issue

Execute an upload with both key that were already upload and new keys. 

## Possible Fix

After debugging EFGS I noticed that the important part of the code where assigning to each status is this ([/service/DiagnosisKeyEntityService](https://github.com/eu-federation-gateway-service/efgs-federation-gateway/blob/master/src/main/java/eu/interop/federationgateway/service/DiagnosisKeyEntityService.java)): 

```java
    diagnosisKeyEntities.forEach(key -> {
      int index = diagnosisKeyEntities.indexOf(key);
      key.setCreatedAt(uploadTimestamp);
      try {
        saveDiagnosisKeyEntity(key);
        resultMap.get(201).add(index);
      } catch (DataIntegrityViolationException e) {
        log.error("{}: {}", index, e.getMessage());
        resultMap.get(409).add(index);
      } catch (Exception e) {
        resultMap.get(500).add(index);
      }
    });

    if (resultMap.get(409).size() > 0 || resultMap.get(500).size() > 0) {
      resultMap.get(201).clear();

      EfgsMdc.put("insertedKeyCount", resultMap.get(201).size());
      EfgsMdc.put("conflictKeysCount", resultMap.get(409).size());
      EfgsMdc.put("failedKeysCount", resultMap.get(500).size());

      log.error("error inserting keys");
      throw new DiagnosisKeyInsertException("Error during insertion of diagnosis keys!", resultMap);
    }
```

It makes sense how each Diagnosis Keys is being assigned to each status, however the clean in [line 97](https://github.com/eu-federation-gateway-service/efgs-federation-gateway/blob/master/src/main/java/eu/interop/federationgateway/service/DiagnosisKeyEntityService.java#L97) seems odd to me. This means that whenever we get either a 409 or a 500 in the keys processing all the 201 keys are wiped out from the response. Even more, the following logs seem to then yield a wrong result:

 ```java
      resultMap.get(201).clear(); // 201 keys are now empty

      EfgsMdc.put("insertedKeyCount", resultMap.get(201).size()); // ALWAYS 0 
      EfgsMdc.put("conflictKeysCount", resultMap.get(409).size());
      EfgsMdc.put("failedKeysCount", resultMap.get(500).size());

      log.error("error inserting keys"); // misleading since some keys were already added
```

Since it's very explicit, it seems to be done on purpose. If that's the case, could you explain why this is needed?

### Implementation suggestions

1. Do not clear 201 results ever `resultMap.get(201).clear()`
2. Move the logs to outside of the `if` block. Makes sense to me to always log all the counts for all statuses. 
3. Change the log message `"error inserting keys"` to `"some keys were not successfully inserted"` or something similar

**Comment by daniel-eder** [comment]
Thank you for the bug report and suggested fix. Our developers will review this and we will get back to you.

**Comment by f11h** [comment]
Hi  @fredrb ,

yes I did this on purpose. A non empty list of 201-keys would suggest that they were actually inserted into database but as per definition only complete batches can be inserted.
So I decided to return an empty 201-list to indicate that no keys were uploaded.

I can understand your intention behind this request. I will consult our architecture team to find a decision how we will proceed with that :)
